tools/ALARAJOYWrapper/preprocess_fendl3.py

```diff
@@ -13,12 +13,84 @@ def args():
     parser.add_argument(
         '--fendlFileDir', '-f', required=False, nargs=1
     )
+    parser.add_argument(
+        '--gas_handling', '-g', required=True, nargs=1
+    )
     parser.add_argument(
         # Temperature for NJOY run [Kelvin]
         '--temperature', '-t', required=False, nargs=1, default=[293.16]
     )
     return parser.parse_args()
 
+def remove_gas_daughters(all_rxns):
+    """
+    Remove reactions from the dictionary that produce a light gas daughter
+        (i.e. a nuclide lighter than an alpha particle) whose total gas
+        production is otherwise accounted for by the MT = 203-207 "reactions"
+        to avoid double-counting.
+
+        Optional method to be called within gas_handling().
+
+    Arguments:
+        all_rxns (collections.defaultdict): Hierarchical dictionary keyed by
+            parent nuclides to store all reaction data, with structured as:
+            {parent:
+                {daughter:
+                    {MT:
+                        {
+                            'emitted': (str of emitted particles)
+                            'non_zero_groups': (int of non-zero groupwise XS)
+                            'xsections': (array of groupwise XS)
+                        }
+                    }
+                }    
+            }
+    
+    Returns:
+        all_rxns (collections.defaultdict): Modified version of all_rxns with
+            double-counted gas-producing reactions left out.
+    """
+
+    gas_MTs = set(tp.GAS_DF['total_mt'])
+    for parent in all_rxns:
+        for daughter in all_rxns[parent]:
+            MTs = list(all_rxns[parent][daughter])
+            if any(MT in gas_MTs for MT in MTs):
+                for MT in MTs:
+                    if MT not in gas_MTs:
+                        del all_rxns[parent][daughter][MT]
```

Comment:
set operations could make this cleaner
```suggestion
            MTs = set(all_rxns[parent][daughter])
            if MTs.intersection(gas_MTs):
                for (MT in MTs - gas_MTs):
                    del all_rxns[parent][daughter][MT]
```